services/cd-service/pkg/service/batch_test.go

```diff
@@ -145,29 +205,30 @@ func TestBatchServiceWorks(t *testing.T) {
 				t.Fatal(err)
 			}
 			for _, tr := range tc.Setup {
-				if err := repo.Apply(testutil.MakeTestContext(), tr); err != nil {
+				if err := repo.Apply(tc.context, tr); err != nil {
 					t.Fatal(err)
 				}
 			}
-			svc := &BatchServer{
-				Repository: repo,
-			}
-			resp, err := svc.ProcessBatch(
-				testutil.MakeTestContext(),
+			tc.svc.Repository = repo
+			resp, err := tc.svc.ProcessBatch(
+				tc.context,
 				&api.BatchRequest{
 					Actions: tc.Batch,
 				},
 			)
+			if err != nil && err.Error() == tc.expectedError {
+				return
+			}
```

Comment:
Please invert this and do not just "return"
```suggestion
			if err != nil && err.Error() != tc.expectedError {
				t.Fatal(err.Error())
			}
```